app/core/src/test/java/com/fsck/k9/cache/ExpiringCacheTest.kt

```diff
@@ -0,0 +1,171 @@
+package com.fsck.k9.cache
+
+import kotlin.test.Test
+import kotlin.test.assertFalse
+import kotlin.test.assertIs
+import kotlin.test.assertNull
+import kotlin.test.assertSame
+import kotlin.test.assertTrue
+
+class ExpiringCacheTest {
+
+    @Test
+    fun implements_cache_interface() {
+        assertIs<Cache<String, String>>(ExpiringCache<String, String>())
```

Comment:
What's the benefit of this test? Wouldn't we get compile time errors if this didn't hold?

Reply:
The test is acting on the concrete `ExpiringCache` and this test ensures that it still implements the desired `Cache` interface.

Reply:
Otherwise you could change or remove the interface without noticing.

Reply:
At some point we'll have code that creates an `ExpiringCache` instance but returns it as `Cache`. If at some later point `ExpiringCache` no longer implements the interface, this will be a compile-time error. If we never use the interface anywhere, there won't be a compile-time error, but the interface won't be useful/necessary either.

Reply:
The interface acts as a contract that allows layering of different caches together and this test ensures that it follows that contract and interoperability.

Reply:
It feels like a test that you can find when using dynamically typed languages. This is something that can and should be "tested" at compile time. The proposed `CacheTest` would accomplish that.

Reply:
If you prefer compile time instead of a dedicated test, then the test could be rearanged to qurantee that without a dedicated test.